packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -0,0 +1,423 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import StoreKit
+
+#if os(iOS)
+  import Flutter
+#elseif os(macOS)
+  import FlutterMacOS
+#endif
+
+public class InAppPurchasePlugin: NSObject, FlutterPlugin, InAppPurchaseAPI {
+  private let receiptManager: FIAPReceiptManager
+  private var productsCache: NSMutableDictionary = [:]
+  private var paymentQueueDelegateCallbackChannel: FlutterMethodChannel?
+  // note - the type should be FIAPPaymentQueueDelegate, but this is only available >= iOS 13,
+  // FIAPPaymentQueueDelegate only gets set/used in registerPaymentQueueDelegateWithError or removePaymentQueueDelegateWithError, which both are ios13+ only
+  private var paymentQueueDelegate: Any?
+  private var requestHandlers = Set<FIAPRequestHandler>()
+  private var handlerFactory: ((SKRequest) -> FIAPRequestHandler)
+  // TODO(louisehsu): Once tests are migrated to swift, we can use @testable import, and make theses vars private again and remove all instances of @objc
+  @objc
+  public var registrar: FlutterPluginRegistrar?
+  @objc
+  // This property is optional, as it requires self to exist to be initialized.
+  public var paymentQueueHandler: FIAPaymentQueueHandler?
+  @objc
+  // This property is optional, as it needs to be set during plugin registration, and can't be directly initialized.
+  public var transactionObserverCallbackChannel: FlutterMethodChannel?
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    #if os(iOS)
+      let messenger = registrar.messenger()
+    #endif
+    #if os(macOS)
+      let messenger = registrar.messenger
+    #endif
+    let channel = FlutterMethodChannel(
+      name: "plugins.flutter.io/in_app_purchase",
+      binaryMessenger: messenger)
+    let instance = InAppPurchasePlugin(registrar: registrar)
+    registrar.addMethodCallDelegate(instance, channel: channel)
+    registrar.addApplicationDelegate(instance)
+    SetUpInAppPurchaseAPI(messenger, instance)
+  }
+
+  @objc
+  // This is the designated initializer
```

Comment:
An init without `convenience` is designated initializer, so no need to explain this. Can you comment when should this init be used instead?